deploy/webhook.yaml

```diff
@@ -0,0 +1,117 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: webhook-metrics-reader
+rules:
+- nonResourceURLs:
+  - /metrics
+  verbs:
+  - get
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: webhook-proxy-role
```

Comment:
Is this ClusterRole needed?

Reply:
The proxy role is needed. Without this not working

Reply:
Where is the proxy used?

Reply:
This question still applies. Where is the validating webhook checking user access?

Reply:
Kubebuilder name it as webhook-proxy-role. I tested removed cluster-role. Webhook then result is fail with 
```
 Internal error occurred: failed calling webhook "policy.open-cluster-management.io.webhook": failed to call webhook: Post "https://webhook-webhook-service.open-cluster-management.svc:443/validate-policy-open-cluster-management-io-v1-policy?timeout=10s": dial tcp 10.96.119.57:443: connect: connection refused
```

Hard to find a doc that explains why kubebuilder made this.
I will look for this more